hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/TableIterator.java

```diff
@@ -54,4 +56,50 @@ public interface TableIterator<KEY, T> extends Iterator<T>, Closeable {
    */
   void removeFromDB() throws RocksDatabaseException, CodecException;
 
+  /**
+   * Convert the given {@link KeyValueIterator} to a {@link TableIterator} using the given converter.
+   *
+   * @param <K> The key type of both the input and the output iterators
+   * @param <INPUT> The value type of the input iterator
+   * @param <OUTPUT> The value type of the output iterator
+   */
+  static <K, INPUT, OUTPUT> TableIterator<K, OUTPUT> convert(KeyValueIterator<K, INPUT> i,
```

Comment:
It would be nice to have just `Iterator<OUTPUT>` returned by `keyIterator` and `valueIterator` methods.
```suggestion
  static <K, INPUT, OUTPUT> Iterator<OUTPUT> convert(KeyValueIterator<K, INPUT> i,
```

Reply:
No, it can't since the returned iterator needs a close() method for releasing resources.